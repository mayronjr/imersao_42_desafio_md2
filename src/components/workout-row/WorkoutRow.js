import 'bootstrap/dist/css/bootstrap.min.css';

function WorkoutRow({ item }) {
    if(item !== undefined){
        return (
            <div className={"row-" + item.id} data-testid={"row-" + item.id}>
                <div className="col-sm-4" >{item.tempo}</div>
                <div className="col-sm-4" >{item.type}</div>
                <div className="col-sm-4" >{item.date.toLocaleDateString()}</div>
            </div>
        )
    }else{
        return null
    }
}

export default WorkoutRow;