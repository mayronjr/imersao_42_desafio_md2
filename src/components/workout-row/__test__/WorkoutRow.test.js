import React from 'react'
import ReactDOM from 'react-dom'
import WorkoutRow from '../WorkoutRow.js'

import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"

import renderer from 'react-test-renderer'

const list = [
    { tempo: 1, type: 'Run', date: new Date(2011, 10, 24) },
    { tempo: 3, type: 'Swimming', date: new Date(2011, 8, 24) }
]

list[0].date.toLocaleDateString()

// afterEach(cleanup)

it('renders without crashing', ()=>{
    const div = document.createElement('div')
    ReactDOM.render(<WorkoutRow/>, div)
})

it('renders Counter Correctly', ()=>{
    const {getByTestId} = render(<WorkoutRow item={list[0]} id={0}/>)
    expect(getByTestId('row-0')).toHaveTextContent("1Run11/24/2011")
})

it('renders Counter Correctly', ()=>{
    const {getByTestId} = render(<WorkoutRow  item={list[1]} id={1}/>)
    expect(getByTestId('row-1')).toHaveTextContent("3Swimming9/24/2011")
})

it('matches snapshot', ()=>{
    const tree = renderer.create(<WorkoutRow counter={3}/>).toJSON()
    expect(tree).toMatchSnapshot()
})