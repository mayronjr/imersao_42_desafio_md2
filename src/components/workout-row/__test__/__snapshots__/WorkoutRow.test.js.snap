// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="row-1"
  data-testid="row-1"
>
  <div
    className="col-sm-4"
  >
    3
  </div>
  <div
    className="col-sm-4"
  >
    Swimming
  </div>
  <div
    className="col-sm-4"
  >
    9/24/2011
  </div>
</div>
`;
